bounce/db/user.py

```diff
@@ -62,23 +62,46 @@ def select_by_id(session, user_id):
     return session.query(User).filter(User.identifier == user_id).first()
 
 
-def search(session, query=None, page=0, size=MAX_SIZE):
-    """Returns a list of clubs that contain content from the user's query"""
+def search(session,
+           fullname=None,
+           username=None,
+           identifier=None,
+           email=None,
+           created_at=None,
+           page=0,
+           size=MAX_SIZE):
+    """Returns a list of users that contain content from the user's query"""
     # number used for offset is the
     # page number multiplied by the size of each page
+
     offset_num = page * size
     users = session.query(User)
 
-    if query:
-        # show clubs that have a name that matches the query
-        users = users.filter(User.full_name.ilike(f'%{query}%'))
+    not_null_filters = []
+
+    triggered = False
+    if fullname or username or email or identifier or created_at:
+        triggered = True
+    if fullname:
+        not_null_filters.append(User.full_name.ilike(f'%{fullname}%'))
+    if username:
+        not_null_filters.append(User.username.ilike(f'%{username}%'))
+    if email:
+        not_null_filters.append(User.email.ilike(f'%{email}%'))
+    if identifier:
```

Comment:
I'd say don't use `ilike` for the identifier because having similar IDs doesn't really tell us anything. Instead, I think we should check for an exact match on the ID.